temporalio/worker/_nexus.py

```diff
@@ -0,0 +1,475 @@
+                    op_resp = temporalio.api.nexus.v1.StartOperationResponse(
+                        async_success=temporalio.api.nexus.v1.StartOperationResponse.Async(
+                            operation_token=result.token,
+                            links=[
+                                temporalio.api.nexus.v1.Link(url=l.url, type=l.type)
+                                for l in ctx.outbound_links
+                            ],
+                        )
+                    )
+                elif isinstance(result, nexusrpc.handler.StartOperationResultSync):
+                    # TODO(nexus-prerelease): error handling here; what error type should it be?
+                    [payload] = await self._data_converter.encode([result.value])
+                    op_resp = temporalio.api.nexus.v1.StartOperationResponse(
+                        sync_success=temporalio.api.nexus.v1.StartOperationResponse.Sync(
+                            payload=payload
+                        )
+                    )
+                else:
+                    # TODO(nexus-prerelease): what should the error response be when the user has failed to wrap their return type?
+                    # TODO(nexus-prerelease): unify this failure completion with the path above
+                    err = TypeError(
+                        "Operation start method must return either nexusrpc.handler.StartOperationResultSync "
+                        "or nexusrpc.handler.StartOperationResultAsync"
+                    )
+                    handler_err = _exception_to_handler_error(err)
+                    return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                        task_token=task_token,
+                        error=temporalio.api.nexus.v1.HandlerError(
+                            error_type=handler_err.type.value,
+                            failure=await self._exception_to_failure_proto(
+                                handler_err.__cause__
+                            ),
+                            retry_behavior=(
+                                temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_RETRYABLE
+                                if handler_err.retryable
+                                else temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_NON_RETRYABLE
+                            ),
+                        ),
+                    )
+
+                return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                    task_token=task_token,
+                    completed=temporalio.api.nexus.v1.Response(start_operation=op_resp),
+                )
+
+        try:
+            completion = await run()
+            await self._bridge_worker().complete_nexus_task(completion)
+        except Exception:
+            temporalio.nexus.logger.exception("Failed completing Nexus operation")
```

Comment:
Should separate errors that happen during run and errors that happen trying to complete the task. We should always try to complete the task with a failure if there are any failures from `run`.

Reply:
I believe it does that now. PTAL.